crates/feldera-types/src/transport/kafka.rs

```diff
@@ -183,10 +194,59 @@ impl KafkaInputConfig {
         self.set_option_if_missing("group.id", &group_id);
         self.set_option_if_missing("enable.partition.eof", "false");
 
+        let pem_options = self.get_pem_options();
+
+        for (k, v) in pem_options
+            .iter()
+            .map(|(k, v)| (k.replace(".pem", ".location"), save_pem_file(v)))
+        {
+            let v = v?;
+            let v = v.to_string_lossy();
+            self.enforce_option(&k, v.as_ref())?;
+        }
+
+        for (k, _) in pem_options {
+            self.kafka_options.remove(&k);
+        }
+
         Ok(())
     }
 }
 
+/// Saves the PEM file to `current_dir()/[FILE-HASH].pem`
+fn save_pem_file(pem: &str) -> AnyResult<PathBuf> {
+    // try to get the path to the appropriate config directory
+    let mut path = match dirs::config_dir() {
+        Some(p) => p,
+        None => {
+            // fall back to the temp directory
+            log::warn!("config directory not found, using the temp directory instead");
+            std::env::temp_dir()
+        }
+    };
+
+    // recursively create the config directory if it doesn't exist
+    if !path.exists() {
+        std::fs::create_dir_all(&path)?;
+    }
+
+    // hash the pem key for file name
+    let hash = sha2::Sha256::new().chain_update(pem).finalize();
```

Comment:
If multiple connectors are using the same certificates (which is very likely), they will end up with the same hash and will all write the same file. This can cause nasty race conditions. I suppose we should include endpoint id in the hash.

I also like @snkas 's idea to store this in the pipeline directory. @snkas , do you know if this is guaranteed to be the `cwd` of the pipeline (it is with the local runner)?

Reply:
The working directory of the pipeline is where the executable is in/executed and it is not shared with other pipelines. There is not a guarantee that the directory is empty, but sufficiently unique naming is good.

Ah, I see with regards to the filename, good catch @abhizer . The endpoint name is unique though, so can be used to prefix the filename to create a unique filename.